org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultGradleBuild.java

```diff
@@ -73,9 +83,31 @@ public boolean isSyncRunning() {
     }
 
     @Override
-    public ModelProvider getModelProvider() {
-        ModelRepository modelRepository = CorePlugin.modelRepositoryProvider().getModelRepository(this.attributes);
-        return new DefaultModelProvider(modelRepository);
+    public <T> T fetchModel(Class<T> model, CancellationToken token, IProgressMonitor monitor) {
+        ModelBuilder<T> modelBuilder = ConnectionAwareLauncherProxy.newModelBuilder(openConnection(this.attributes), model);
+        applyRequestAttributes(modelBuilder, this.attributes, getTransientRequestAttributes(token, monitor));
+        return modelBuilder.get();
+    }
+
+    @Override
+    public <T> Collection<T> fetchCompositeModel(Class<T> model, CancellationToken token, IProgressMonitor monitor) {
```

Comment:
I'd just call it `fetchModels`, because we might not be talking to a composite at all